bynder_sdk/client/upload_client.py

```diff
@@ -1,144 +1,132 @@
-                'Filename': key.format(part_nr),
-                'key': key.format(part_nr),
-            })
-            return data
-
-        return part_data
-
-    def _register_part(self, init_data, part_nr):
-        """ Registers an uploaded chunk in Bynder.
+        response = self.session.post('/v7/file_cmds/upload/prepare')
+        return response["file_id"]
+
+    def _upload_chunks(self, file_path, file_id):
+        """Upload the file in chunks of MAX_CHUNK_SIZE.
+        :param file_path: The path of the asset to be uploaded.
+        :param file_id: The uuid4 used to identify the file to be uploaded.
+        :return:
+            - chunks_count: The number of chunks in which the file is to
+        be uploaded.
+            - file_size: The size of the file to be uploaded.
         """
+        chunks_count = 0
+        file_size = 0
+        with open(file_path, "rb") as f:
+            file_size = os.stat(f.fileno()).st_size
+            chunks_count = math.ceil(
+                file_size / MAX_CHUNK_SIZE)
+            for chunk_nr, chunk in enumerate(_read_in_chunks(f)):
+                self._upload_chunk(file_id, chunk, chunk_nr)
+
+        return chunks_count, file_size
+
+    def _upload_chunk(self, file_id, data, chunk_nr):
+        self.session.headers['content-sha256'] = sha256(data).hexdigest()
         self.session.post(
-            '/v4/upload/',
-            data={
-                'id': init_data['s3file']['uploadid'],
-                'targetid': init_data['s3file']['targetid'],
-                'filename': init_data['s3_filename'],
-                'chunkNumber': part_nr
-            }
+            '/v7/file_cmds/upload/{}/chunk/{}'.format(file_id, chunk_nr),
+            data=data
         )
 
-    def _finalise_file(self, init_data, total_parts):
-        """ Finalises a completely uploaded file.
+    def _finalise_file(self, file_id, file_name, file_size, chunks_count):
+        """Finalises a completely uploaded file.
+        :param file_id: The file_id returned from the prepare endpoint.
+        :param file_name: The name of the file to be uploaded.
+        :param file_size: The size of the file to be uploaded.
+        :param chunks_count: The number of chunks in which the file was
+        uploaded.
         """
-        return self.session.post(
-            '/v4/upload/{0}/'.format(init_data['s3file']['uploadid']),
+        self.session.post(
+            '/v7/file_cmds/upload/{}/finalise_api'.format(file_id),
             data={
-                'id': init_data['s3file']['uploadid'],
-                'targetid': init_data['s3file']['targetid'],
-                's3_filename': init_data['s3_filename'],
-                'chunks': total_parts
+                'fileName': file_name,
+                'fileSize': file_size,
+                'chunksCount': chunks_count,
+                "sha256": self.file_sha256,
+                "intent": "upload_main_uploader_asset",
```

Comment:
The intent can be left out from the SDK requests. It will fallback to the correct one server side.